File: solutions/1684. Count the Number of Consistent Strings/1684.cpp
// Time complexity: O(m+nk)
// Space complexity: O(1)
class Solution {
 public:
  int countConsistentStrings(string_view allowed, const vector<string>& words) {
    vector<bool> isAllowed(26, false);

    for (int i = 0; i < allowed.length(); ++i)
      isAllowed[allowed[i] - 'a'] = true;

    int ans = 0;

    for (const string_view word : words) {
      bool isConsistent = true;
      for (int i = 0; i < word.length(); ++i)
        if (!isAllowed[word[i] - 'a']) {
          isConsistent = false;
          break;
        }

      if (isConsistent)
        ++ans;
    }

    return ans;
  }
};

File: solutions/1684. Count the Number of Consistent Strings/1684.py
# Time complexity: O(m+nk)
# Space complexity: O(1)
class Solution:
  def countConsistentStrings(self, allowed: str, words: List[str]) -> int:
    isAllowed = [False] * 26

    for char in allowed:
      isAllowed[ord(char) - ord('a')] = True

    ans = 0

    for word in words:
      isConsistent = True

      for char in word:
        if not isAllowed[ord(char) - ord('a')]:
          isConsistent = False
          break

      if isConsistent:
        ans += 1

    return ans